### PR TITLE
Docket switch VLJ ruling

### PR DESCRIPTION
**Pull request by leikkisa**
connects #15336 
connects #15338

### Description
This PR is for a Judge to submit their ruling on a docket switch, and to create the next task for the Clerk of the Board attorney.

### Acceptance Criteria
#15336 
- [ ] Please put this work behind the feature toggle: `docket_switch`
- [ ] This feature should be accessible to the following user groups: `VLJ`
- [ ] When a VLJ selects "Rule on Docket Switch", they are taken to this screen
- [ ] At the top of the screen, they see instructions and information (summary, timeliness, recommendation, hyperlink) from the attorney - please format according to the design below
- [ ] Please format the information from the attorney based on the design (bolded labels, line-break and 24px space after each section)
- [ ] If the URL is a valid URL, please make it clickable
- [ ] The user can select their ruling via the radio buttons (grant all issues, grant a partial switch, deny all issues) and this will trigger either a `Denied Docket Switch` or `Granted Docket Switch` task 
- [ ] The user can enter in text in the hyperlink field - this is an optional field, and they don't need to enter in a valid hyperlink (e.g. they could write in: "sent via email") 
- [ ] The user can enter in context or instructions via the notes field 
- [ ] The user can assign this to a ClerkOfTheBoard attorney - it defaults to the user that assigned the task to them, and they should see all attorney members of the `ClerkOfTheBoard` organization as options (VLJ's excluded)
- [ ] All fields except the hyperlink field are required
- [ ] Include screenshot(s) in the Github issue if there are front-end changes
- [ ] Add storybook story
- [ ] Add jest test
- [ ] Add to rspec feature test for happy path

#15338
- [ ] Please put this work behind the feature toggle: `docket_switch`
- [ ] This feature should be accessible to the following user groups: `ClerkOfTheBoard` organization
- [ ] Assigned user should receive a `Denied Docket Switch` or `Granted Docket Switch` task based on the VLJ's ruling in their queue
- [ ] Task instructions (from the VLJ) should be formatted as (nothing needs to be bolded, add line breaks after the ruling and signed ruling letter) - please see the design for more context: 
  - [ ] I am proceeding with a [full switch/partial switch/denial].
  - [ ] Signed ruling letter: [insert hyperlink/text here] - if this is a valid hyperlink, please make this clickable
  - [ ] [Context/instructions]
- [ ] Include screenshot(s) in the Github issue if there are front-end changes

### Testing Plan
1. Automated tests

### User Facing Changes
 - [x] Screenshots of UI changes added to PR & Original Issue

### Code Documentation Updates
- [ ] Add or update code comments at the top of the class, module, and/or component.

### Storybook Story
*For Frontend (Presentationa) Components*
* [x] Add a [Storybook](https://github.com/department-of-veterans-affairs/caseflow/wiki/Documenting-React-Components-with-Storybook) file alongside the component file (e.g. create `MyComponent.stories.js` alongside `MyComponent.jsx`)
* [x] Give it a title that reflects the component's location within the overall Caseflow hierarchy
* [ ] Write a separate story (within the same file) for each discrete variation of the component

### Database Changes
*Only for Schema Changes*

* [ ] Timestamps (created_at, updated_at) for new tables
* [x] Column comments updated
* [ ] Have your migration classes inherit from `Caseflow::Migration`, especially when adding indexes (use `add_safe_index`)
* [ ] Verify that `migrate:rollback` works as desired ([`change` supported functions](https://edgeguides.rubyonrails.org/active_record_migrations.html#using-the-change-method))
* [ ] Query profiling performed (eyeball Rails log, check bullet and fasterer output)
* [ ] Appropriate indexes added (especially for foreign keys, polymorphic columns, unique constraints, and Rails scopes)
* [ ] DB schema docs updated with `make docs` (after running `make migrate`)
* [x] #appeals-schema notified with summary and link to this PR
* [ ] Any non-obvious semantics or logic useful for interpreting database data is documented at [Caseflow Data Model and Dictionary](https://github.com/department-of-veterans-affairs/caseflow/wiki/Caseflow-Data-Model-and-Dictionary)

### CODE REVIEW

**Comment by leikkisa** [comment]
Screen Shot 2020-11-13 at 7 55 12 AM
Screen Shot 2020-11-09 at 6 18 52 PM

File: app/models/tasks/docket_switch/docket_switch_granted_task.rb [hunk 1]
```diff
@@ -17,5 +17,9 @@ class << self
     def label
       COPY::DOCKET_SWITCH_GRANTED_TASK_LABEL
     end
+
+    def verify_user_can_create!(user, parent)
+      parent.is_a?(DocketSwitchRulingTask) ? true : super(user, parent)
+    end
```
Comment (leikkisa) [inline app/models/tasks/docket_switch/docket_switch_granted_task.rb:23]: This was a bit of an improvisation, I didn't see anything specific about the task creation permissions, but by default, we couldn't create the granted or denied tasks unless they were listed as available actions for the judge, or something along those lines.

I think the only place where those can get created right now is from the ruling task, which only judges can do.

File: client/app/queue/docketSwitch/recommendDocketSwitch/RecommendDocketSwitchForm.jsx [hunk 2]
```diff
@@ -12,7 +12,7 @@ import {
 import AppSegment from '@department-of-veterans-affairs/caseflow-frontend-toolkit/components/AppSegment';
 import Button from '../../../components/Button';
 import { sprintf } from 'sprintf-js';
-import { dispositions } from '../constants';
+import DISPOSITIONS from '../../../../constants/DOCKET_SWITCH';
```
Comment (jcq) [inline client/app/queue/docketSwitch/recommendDocketSwitch/RecommendDocketSwitchForm.jsx:15]: if we're moving things this far up in the file tree, probably worth switching to the absolute naming syntax:
```jsx
import DISPOSITIONS from 'constants/DOCKET_SWITCH'
```

File: spec/feature/queue/docket_switch_spec.rb [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+        parent: root_task,
+        assigned_to: judge,
+        assigned_by: cotb_attorney
+      )
+    end
+    let(:context) { "Lorem ipsum dolor sit amet, consectetur adipiscing elit" }
+    let(:hyperlink) { "https://example.com/file.txt" }
+
+    # Checks granted, partially_granted, and denied dispositions
+    Constants::DOCKET_SWITCH.each_key do |disposition|
```
Comment (jcq) [inline spec/feature/queue/docket_switch_spec.rb:119]: Very comprehensive!

**Review by jcq — APPROVED** [review]
Looks great! Love the added stories, tests, etc

And thank you for updating the rest of the stuff from "change" to "switch"

**Comment by yoomlam** [comment]
From Ted Mills (TIB):
> The question is ‘how is docket switching represented in the database to facilitate aggregated, not just individual case, reporting?’.  The law requires that the Board provide mass reporting on this and not just facilitate individual cases.  We don’t know how this is intended to be reflected in the database, so we need some help in getting there.

You can use the `docket_switches` table, which has references the old and new AMA appeals (in the `appeals` table) via the `old_docket_stream_id` and `new_docket_stream_id`.

**Comment by vacohitchs** [comment]
@yoomlam : docket_switches.old_docket_stream_id  and new_docekt_stream_id represent appeal.id?
What does docket_switches.receipt_date represent?
Thanks! 
@HelimaBass 
@TedM-BVA

**Comment by jcq** [comment]
@vacohitchs We can update the schema with a comment that clarifies that. In the meantime, it is the value entered by the user during the docket switch process ([see UI](https://www.caseflowdemo.com/storybook/?path=/story/queue-docket-switch-docketswitchreviewrequestform--basic)). It is the date the Board received the second NOD (the one that requested the docket switch).

**Pull request merged**